### PR TITLE
Allow partially read chapters to be marked as unread in updates screen

### PR DESCRIPTION
**Pull request by zbue**

### CODE REVIEW

File: app/src/main/java/eu/kanade/tachiyomi/ui/updates/UpdatesScreenModel.kt [hunk 1]
```diff
@@ -110,24 +112,32 @@ class UpdatesScreenModel(
...
                 downloadProgressProvider = { activeDownload?.progress ?: 0 },
-                selected = it.chapterId in selectedChapterIds,
+                selected = update.chapterId in selectedChapterIds,
+                readProgressString = update.lastPageRead
+                    .takeIf { !update.read && it > 0 }
+                    ?.let {
+                        context.getString(
+                            R.string.chapter_progress,
+                            it + 1,
+                        )
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/ui/updates/UpdatesScreenModel.kt:139]: This should be happening in the composable, not the screen model.

Comment (zbue) [reply app/src/main/java/eu/kanade/tachiyomi/ui/updates/UpdatesScreenModel.kt:139]: Should I do it for manga screen as well?

Comment (arkon) [reply app/src/main/java/eu/kanade/tachiyomi/ui/updates/UpdatesScreenModel.kt:139]: Sure, but I don't mind having that change seperate

File: app/src/main/java/eu/kanade/presentation/updates/UpdatesUiItem.kt [hunk 2]
```diff
@@ -113,6 +117,14 @@ fun LazyListScope.updatesUiItems(
                     modifier = Modifier.animateItemPlacement(),
                     update = updatesItem.update,
                     selected = updatesItem.selected,
+                    readProgress = updatesItem.update.lastPageRead
+                        .takeIf { !updatesItem.update.read && it > 0L }
+                        ?.let {
+                            context.getString(
```
Comment (arkon) [inline app/src/main/java/eu/kanade/presentation/updates/UpdatesUiItem.kt:123]: Can just use `stringResource` in composables.

File: app/src/main/java/eu/kanade/tachiyomi/ui/manga/MangaScreenModel.kt [hunk 3]
```diff
@@ -507,12 +507,14 @@ class MangaInfoScreenModel(
                             dateFormat,
                         )
                     },
-                readProgressString = chapter.lastPageRead.takeIf { !chapter.read && it > 0 }?.let {
-                    context.getString(
-                        R.string.chapter_progress,
-                        it + 1,
-                    )
-                },
+                readProgressString = chapter.lastPageRead
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/ui/manga/MangaScreenModel.kt:510]: Set the number and do the string part in the composable.

**Review by arkon — CHANGES_REQUESTED** [review]

**Pull request merged**